controllers/metal3machine_controller_test.go

```diff
@@ -308,75 +307,90 @@ var _ = Describe("Metal3Machine manager", func() {
 		)
 	})
 
-	// Legacy tests
-	It("TestMetal3MachineReconciler_Metal3ClusterToMetal3Machines", func() {
-		metal3Cluster := newMetal3Cluster("my-metal3-cluster",
-			bmcOwnerRef(), bmcSpec(), nil, false,
-		)
-		objects := []runtime.Object{
-			newCluster(clusterName, nil, nil),
-			metal3Cluster,
-			newMachine(clusterName, "my-machine-0", "my-metal3-machine-0"),
-			newMachine(clusterName, "my-machine-1", "my-metal3-machine-1"),
-			// Intentionally omitted
-			newMachine(clusterName, "my-machine-2", ""),
-		}
-		c := fake.NewFakeClientWithScheme(setupScheme(), objects...)
-		r := Metal3MachineReconciler{
-			Client: c,
-			Log:    klogr.New(),
-		}
-		mo := handler.MapObject{
-			Object: metal3Cluster,
-		}
-		out := r.Metal3ClusterToMetal3Machines(mo)
-		machineNames := make([]string, len(out))
-		for i := range out {
-			machineNames[i] = out[i].Name
-		}
-		Expect(len(out)).To(Equal(2), "Expected 2 Metal3 machines to reconcile but got %d", len(out))
+	type TestCaseMetal3ClusterToBMM struct {
+		Cluster       *capi.Cluster
+		M3Cluster     *infrav1.Metal3Cluster
+		Machine0      *capi.Machine
+		Machine1      *capi.Machine
+		Machine2      *capi.Machine
+		M3Machine0    *infrav1.Metal3Machine
+		ExpectRequest int
```

Comment:
I would understand 'ExpectedRequest' if it was 'bool'.
A comment would clarify the usage now when the type is 'int'. Below values 0,1,2 seem to be used. The meaning?

Reply:
The number of requests returned by the function.

Reply:
Changed it to bool.